README.md

```diff
@@ -1,2 +1,69 @@
+- [Description](#description)
+- [Bandwidth](#bandwidth)
+- [Environmental Variables](#environmental-variables)
+- [Callback URLs](#callback-urls)
+    - [Ngrok](#ngrok)
+
+<!-- /TOC -->
+
+# Description
+Using a tool capable of making POST requests (Postman), send a POST request to the app's endpoint `/callbacks/outbound/messaging` with a json body like:
+```json
+{
+  "to": "+19199994444",
+  "text":"Hello World!"
+}
+```
+The application will text the number `+19199994444` a picture of a cat and the words `Hello World!`.
+If you text your Bandwidth number any media file after setting up the application's callback in the bandwidth dashboard; it will save it to the project folder.
+
+The other two endpoints are used for handling inbound and outbound webhooks from Bandwidth. In order to use the correct endpoints, you must check the "Use multiple callback URLs" box on the application page in Dashboard. Then in Dashboard, set the INBOUND CALLBACK to /callbacks/inbound/messaging and the STATUS CALLBACK to /callbacks/outbound/messaging/status. The same can be accomplished via the Dashboard API by setting InboundCallbackUrl and OutboundCallbackUrl respectively.
+
+Inbound callbacks are sent notifying you of a received message on a Bandwidth number, this app saves an inbound image to the working directory. Outbound callbacks are status updates for messages sent from a Bandwidth number, this app has a dedicated response for each type of status update.
+
+# Bandwidth
+
+In order to use the Bandwidth API users need to set up the appropriate application at the [Bandwidth Dashboard](https://dashboard.bandwidth.com/) and create API tokens.
+
+To create an application log into the [Bandwidth Dashboard](https://dashboard.bandwidth.com/) and navigate to the `Applications` tab.  Fill out the **New Application** form selecting the service (Messaging or Voice) that the application will be used for.  All Bandwidth services require publicly accessible Callback URLs, for more information on how to set one up see [Callback URLs](#callback-urls).
+
+For more information about API credentials see [here](https://dev.bandwidth.com/guides/accountCredentials.html#top)
+
+# Environmental Variables
+The sample app uses the below environmental variables.
+```sh
+BW_ACCOUNT_ID                        # Your Bandwidth Account Id
+BW_USERNAME                          # Your Bandwidth API Username
+BW_PASSWORD                          # Your Bandwidth API Password
+BW_MESSAGING_APPLICATION_ID          # Your Messaging Application Id created in the dashboard
+BW_NUMBER                            # The Bandwidth phone number involved with this application
+LOCAL_PORT                           # The port number you wish to run the sample on
+```
+
+# Callback URLs
+
+For a detailed introduction to Bandwidth Callbacks see https://dev.bandwidth.com/guides/callbacks/callbacks.html
+
+Below are the callback paths:
+* `/callbacks/outbound/messaging` For Sending Media Messages
```

Comment:
```suggestion

```
This isn't an actual callback that a client would need to implement, it's just an example endpoint we're creating, so we shouldn't talk about it like it's some callback URL for our API

Reply:
i agree, but this is the same endpoint used for all of our other sample apps, and they all document it this way, so we'd probably have to change them all to be consistent